mermin/src/span/producer.rs

```diff
@@ -270,89 +280,119 @@ impl FlowSpanProducer {
         let worker_count = self.worker_count.max(1);
 
         loop {
-            let mut guard = match async_fd.readable().await {
-                Ok(guard) => guard,
-                Err(e) => {
-                    error!(
-                        event.name = "span.producer.error",
-                        error.message = %e,
-                        "error waiting for ring buffer readability"
-                    );
-                    break;
-                }
-            };
-
-            while let Some(item) = flow_events.next() {
-                let timer = metrics::registry::PROCESSING_LATENCY
-                    .with_label_values(&["flow_ingestion"])
-                    .start_timer();
-
-                let flow_event: FlowEvent =
-                    unsafe { std::ptr::read_unaligned(item.as_ptr() as *const FlowEvent) };
-
-                metrics::userspace::inc_ringbuf_packets("received", 1);
-                metrics::userspace::inc_ringbuf_bytes(flow_event.snaplen as u64);
-
-                let mut sent = false;
-                for attempt in 0..worker_count {
-                    let current_worker = (worker_index + attempt) % worker_count;
-                    let worker_tx = &worker_channels[current_worker];
-
-                    match worker_tx.try_send(flow_event) {
-                        Ok(_) => {
-                            metrics::userspace::inc_channel_sends("packet_worker", "success");
-                            worker_index = (current_worker + 1) % worker_count;
-                            sent = true;
+            tokio::select! {
+            _ = shutdown_rx.recv() => {
+                info!(
```

Comment:
Since there is bunch of other places that we log that a shutdown is received, this is really only needed for developer tracing so I'd make this trace level or just remove it.